File: java-codegen/src/main/java/org/opensearch/client/codegen/core/Naming.java
package org.opensearch.client.codegen.core;

import org.opensearch.client.codegen.utils.Strings;
import software.amazon.smithy.model.shapes.MemberShape;
import software.amazon.smithy.model.shapes.OperationShape;
import software.amazon.smithy.model.shapes.ServiceShape;
import software.amazon.smithy.model.shapes.StructureShape;

public class Naming {
    public static String defaultName(MemberShape shape) {
        return Strings.toCamelCase(shape.getMemberName());
    }

    public static String defaultName(OperationShape shape) {
        return Strings.toCamelCase(shape.getId().getName());
    }

    public static String defaultName(ServiceShape shape) {
        return Strings.toPascalCase(shape.getId().getName()) + "Client";
    }

    public static String asyncName(ServiceShape shape) {
        return Strings.toPascalCase(shape.getId().getName()) + "AsyncClient";
    }

    public static String defaultName(StructureShape shape) {
        return Strings.toPascalCase(shape.getId().getName());
    }
}

File: java-codegen/src/main/kotlin/org/opensearch/client/codegen/core/Naming.kt
package org.opensearch.client.codegen.core

import org.opensearch.client.codegen.utils.toCamelCase
import org.opensearch.client.codegen.utils.toPascalCase
import software.amazon.smithy.model.shapes.MemberShape
import software.amazon.smithy.model.shapes.OperationShape
import software.amazon.smithy.model.shapes.ServiceShape
import software.amazon.smithy.model.shapes.StructureShape

fun MemberShape.defaultName(): String = memberName.toCamelCase()

fun OperationShape.defaultName(): String = id.name.toPascalCase()

fun ServiceShape.defaultName(): String = id.name.toPascalCase() + "Client"

fun ServiceShape.asyncName(): String = id.name.toPascalCase() + "AsyncClient"

fun StructureShape.defaultName(): String = id.name.toPascalCase()